qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/storage/QEPDataset.java

```diff
@@ -261,7 +266,7 @@ public void setComponentInDelta(TripleComponentRole role, long component) {
 	public QueryCloseableIterator search(CharSequence subject, CharSequence predicate, CharSequence object)
 			throws QEPCoreException {
 		QEPDatasetContext ctx = createContext();
-		return search(ctx, subject, predicate, object).attach(ctx);
+		return (QueryCloseableIterator) search(ctx, subject, predicate, object).attach(ctx);
 	}
 
```

Comment:
This may not be the correct fix, but it was the best I could manage when migrating to RDF4J 5.0.0.